pyblish/plugin.py

```diff
@@ -780,6 +782,44 @@ def current_host():
     return pyblish._registered_hosts[-1] or "unknown"
 
 
+def register_callback(signal, callback):
+    """Register a new callback
+
+    Arguments:
+        signal (string): Name of signal to register the callback with.
+        callback (func): Function to execute when a signal is emitted.
+    """
+
+    if signal in pyblish._registered_callbacks:
+        pyblish._registered_callbacks[signal].append(callback)
+    else:
+        pyblish._registered_callbacks[signal] = [callback]
+
+
+def deregister_callback(signal, callback):
+    """Deregister a callback
+
+    Arguments:
+        signal (string): Name of signal to deregister the callback with.
+        callback (func): Function to execute when a signal is emitted.
+    """
+
+    if callback in pyblish._registered_callbacks[signal]:
+        pyblish._registered_callbacks[signal].remove(callback)
+
+
+def deregister_all_callbacks():
+    """Deregisters all callback"""
+
+    pyblish._registered_callbacks = dict()
```

Comment:
Rather than creating a new dictionary and re-assigning it, it can be good to call `.clear()` instead. That way, anyone holding a reference to that dictionary won't be invalidated.